### DISCUSSION TITLE
Relation property filter with AND doesn't work

### DISCUSSION THREAD

**Discussion by tobycm** [pocketbase/pocketbase#3742] [Q&A] [answered]
![image](https://github.com/pocketbase/pocketbase/assets/62174797/eff524bc-9e31-473e-93ea-686f60ab6a37)

Filter: `colors.name ?= "white" && colors.name ?= "black"`
No result when listing with this filter but there is a product that matches it

Thank you for making this awesome project

**Comment by ganigeorgiev** [comment, marked as answer]
Something similar was reported previously but I'm not sure how to explain it better.

In short it works correctly because internally when you filter by relation fields we perform a `JOIN` with the related collection(s) and `colors.name` will refer to a single record of that join, aka. `colors.name` cannot be 2 different things at the same time.
Internally the above filter will produce a pseudo-query like:

```sql
SELECT example.*
FROM example
LEFT JOIN colors ON ...
WHERE colors.name = "white" AND colors.name = "black"
```

We currently don't support `IN` like operation on nested multiple relation fields, if that's what you are looking for (_I've added a note in the roadmap to consider it together with some other planned filter tasks_).

**Reply by tobycm** [reply]
Oh ok thank you for your fast reply
